make_european.py

```diff
@@ -0,0 +1,82 @@
+"""Take input in the format:
+mm/dd/yyyy:username@host
+Print out all of the usernames that have the host 'aol' and convert
+the date to European format so that the following format is produced:
+dd/mm/yyyy username
+"""
+import re
+
+def get_input():
+    """Prompts the user for input in the format:
+    mm/dd/yyyy:username@host
+    Receive the input as raw_input.
+    """
+    try:
+        return raw_input()
+    except EOFError:
+        return ' '
+
+def convert_input(string, desired_host):
+    """Take in a string and a desired host. If the host found in the
+    string matches the desired host, then convert the found date to the
+    European format and convert the username to all lowercase. Return
+    the new date and username.
+    """
+    assert isinstance(string, str), "Initial input for convert_input needs to be a string."
+    assert isinstance(desired_host, str), "Host variable for convert_input needs to be a string."
+
+    username, host = find_email_address(string)
+    if host.lower() == desired_host:
+        month, day, year = find_date(string)
+        euro_date = '%s/%s/%s' % (day, month, year) # Convert the date into European format.
+        return '%s %s' % (euro_date, username.lower())
+
+def find_email_address(string):
+    """Take in a string and return the username and host of the email
+    address found in the string.
+    """
+    assert isinstance(string, str), "The input for find_email_address needs to be a string."
+
+    email_address = re.search(r'([\w.-]+)@([\w.-]+)', string)
+    assert email_address, "A proper email address must be provided."
+
+    username = email_address.group(1)
+    host = email_address.group(2)
+
+    return (username, host)
+
+def find_date(string):
+    """Take in a string and return the month, day, and year of the date
+    found in the string."""
+    assert isinstance(string, str), "The function find_date takes a string."
+
+    date = re.search('([0-9]{1,2})/([0-9]{1,2})/([0-9]{4})', string)
```

Comment:
playing around with whitespacing can make code, especially terse stuff like regex's, much more readable. i *think* this formatting will work but you may have to toy with it - change this and other regexes to use the following formatting (i'll give this one as an example)

```
date = re.search(
    "([0-9]{1,2})" #the year
    "/([0-9]{1,2})" #the month
    "/([0-9]{4})", #the day
    string
)

if you combine the above with named groups, the comments become redundant. but there are times when something has semantic meaning but you don't want to capture it (so don't need a named group) and comments can be helpful there. i *believe* python auto concatenates all strings that are next to each other so it turns "hello " "world" into "hello world", even over line breaks, which is why the above formatting works (or should).